encrypted-media/util/drm-messagehandler.js

```diff
@@ -0,0 +1,230 @@
+// Expect utf8decoder and utf8decoder to be TextEncoder('utf-8') and TextDecoder('utf-8') respectively
+//
+// drmconfig format:
+// { <keysystem> : {    "serverURL"             : <the url for the server>,
+//                      "httpRequestHeaders"    : <map of HTTP request headers>,
+//                      "servertype"            : "microsoft" | "drmtoday",                 // affects how request parameters are formed
+//                      "certificate"           : <base64 encoded server certificate> } }
+//
+
+drmconfig = {
+    "com.widevine.alpha": {
+        "serverURL": "https://lic.staging.drmtoday.com/license-proxy-widevine/cenc/",
+        "servertype" : "drmtoday",
+        "userId" : "12345",
+        "merchant" : "cablelabs"
+    },
```

Comment:
I'll send you the cert.

Widevine requires a certificate if client identifying data is provided. This depends on the platform and/or requested functionality.

There are simple `setServerCertificate` tests in the Google directory. If we wanted to test that a certificate is required and/or works, we'd probably have to test for the request then on a separate `MediaKeys` provide the certificate and see that we don't get the request. It sounds like this is only possible with Widevine, though.